components/ws-manager/pkg/manager/headless.go

```diff
@@ -172,6 +174,13 @@ func (hl *HeadlessListener) listenAndRetry(ctx context.Context, pod *corev1.Pod,
 		return err
 	}
 
+	logger := log.WithContext(ctx).WithFields(owi).Logger
```

Comment:
I reckon `ghost` workspaces should not have a headless log listener attached at all.

Reply:
see https://github.com/gitpod-io/gitpod/pull/3776